azurelinuxagent/common/event.py

```diff
@@ -393,52 +529,106 @@ def _clean_up_message(message):
-                              DefaultParameter("KeywordName", '')]
+        params_to_keep = dict().fromkeys(['Name', 'Version', 'Operation', 'OperationSuccess', 'Message', 'Duration'])
+        trimmed_params = []
+
+        for param in event.parameters:
+            if param.name in params_to_keep:
+                trimmed_params.append(param)
 
-        # Converting the event_parameters into a dictionary as it helps to easily look up and get values
-        param_names = OrderedDict([(param.name, param.value) for param in event_parameters])
+        event.parameters = trimmed_params
+
+    def collect_events(self):
+        """
+        Retuns a list of events that need to sent to the telemetry pipeline and deletes the corresponding files
+        from the events directory.
+        """
+        event_list = TelemetryEventList()
+        event_directory_full_path = os.path.join(conf.get_lib_dir(), EVENTS_DIRECTORY)
+        event_files = os.listdir(event_directory_full_path)
 
-        for param in default_parameters:
-            if param.name not in param_names or set_values_for_agent:
-                # If set_values_for_agent, we disregard any values already set for an existing default property and
-                # replaces it with a latest entry.
-                param_names[param.name] = param.value
+        for event_file in event_files:
+            try:
+                match = EVENT_FILE_REGEX.search(event_file)
+                if match is None:
+                    continue
 
-        parameters = DataContractList(TelemetryEventParam)
-        for name, value in param_names.items():
-            parameters.append(TelemetryEventParam(name, value))
+                event_file_path = os.path.join(event_directory_full_path, event_file)
 
-        return parameters
+                try:
+                    logger.verbose("Processing event file: {0}", event_file_path)
+
+                    with open(event_file_path, "rb") as fd:
+                        event_data = fd.read().decode("utf-8")
+
+                    event = parse_event(event_data)
+
+                    is_legacy_event = match.group('agent_event') is None
+
+                    if is_legacy_event:
+                        # We'll use the file creation time for the event's timestamp
+                        event_file_creation_time_epoch = os.path.getmtime(event_file_path)
+                        event_file_creation_time = datetime.fromtimestamp(event_file_creation_time_epoch)
+
+                        if event.is_extension_event():
+                            EventLogger._trim_extension_event_parameters(event)
+                            self._add_common_event_parameters(event, event_file_creation_time)
+                        else:
+                            self._update_old_event_schema(event, event_file_creation_time)
+
+                    event_list.events.append(event)
+                finally:
+                    os.remove(event_file_path)
```

Comment:
Suggestion: before this change, we had logged when we were done with an event (basically, when we deleted it):

> logger.verbose("Processed event file: {0}", evt_file_name)

I found it useful when debugging.

Reply:
Yes, this was intentional. We already log "Processing event file:" at line 588 so having a "I'm starting" and a "I completed" for each event is way too verbose.